libcaliptra/examples/generic/test.c

```diff
@@ -854,75 +939,104 @@ int rom_test_devid_csr(const test_info* info)
-    dpe_req.data_size = sizeof(struct dpe_derive_context_cmd);
+    if (caliptra_verify_ecdsa_signature(&exported_resp)) {
+        printf("Sign with exported Ecdsa: OK\n");
+    } else {
+        printf("Error invalid signature.\n");
+        return 1;
+    }
 
-    struct caliptra_invoke_dpe_resp dpe_resp = { 0 };
-    status = caliptra_invoke_dpe_command(&dpe_req, &dpe_resp, false);
+    return 0;
+}
 
-    if (status) {
-        printf("DPE Command failed: 0x%x\n", status);
+// Test exported cdi with a new measurement
+int sign_with_exported_ecdsa_cdi_hitless(const test_info* info)
+{
+    struct dpe_derive_context_response derive_resp = { 0 };
+    struct dpe_derive_context_exported_cdi_response exported_resp = { 0 };
+    struct caliptra_revoke_exported_cdi_handle_req revoke_req = { 0 };
+
+    int status = boot_to_ready_for_fw(info, false);
+
+    if (status){
         dump_caliptra_error_codes();
-        failure = 1;
-    } else {
-        printf("DPE Command: OK\n");
+        return 1;
     }
 
-    struct dpe_derive_context_exported_cdi_response* exported_resp = &dpe_resp.derive_context_exported_cdi_resp;
+    status = caliptra_upload_fw(&info->image_bundle, false);
 
-    // SHA384 of a 48 bytes of 0s
-    uint8_t tbs[] = {
-        0x8f, 0x0d, 0x14, 0x5c, 0x03, 0x68, 0xad, 0x6b, 0x70, 0xbe,
-        0x22, 0xe4, 0x1c, 0x40, 0x0e, 0xea, 0x91, 0xb9, 0x71, 0xd9,
-        0x6b, 0xa2, 0x20, 0xfe, 0xc9, 0xfa, 0xe2, 0x5a, 0x58, 0xdf,
-        0xfd, 0xaa, 0xf7, 0x2d, 0xbe, 0x8f, 0x67, 0x83, 0xd5, 0x51,
-        0x28, 0xc9, 0xdf, 0x4e, 0xfa, 0xf6, 0xf8, 0xa7
-    };
+    if (status)
+    {
+        printf("FW Load Failed: 0x%x\n", status);
+        dump_caliptra_error_codes();
+        return 1;
+    }
 
-    struct caliptra_sign_with_exported_ecdsa_req sign_req = { 0 };
-    memcpy(&sign_req.exported_cdi_handle, exported_resp->exported_cdi_handle, sizeof(uint8_t) * 32);
-    memcpy(&sign_req.tbs, &tbs, sizeof(uint8_t) * 48);
+    // Create first export cdi and certificate
+    if(derive_context_exported_cdi(&exported_resp, DPE_DERIVE_CONTEXT_FLAG_EXPORT_CDI | DPE_DERIVE_CONTEXT_FLAG_CREATE_CERTIFICATE)) {
+        printf("Failed to export first CDI\n");
+        return 1;
+    }
 
-    struct caliptra_sign_with_exported_ecdsa_resp sign_resp = { 0 };
-    status = caliptra_sign_with_exported_ecdsa(&sign_req, &sign_resp, false);
+    if (caliptra_verify_ecdsa_signature(&exported_resp)) {
+        printf("Sign with exported Ecdsa: OK\n");
+    } else {
+        printf("Error invalid signature.\n");
+        return 1;
+    }
 
-    if (status) {
-        printf("Sign with exported Ecdsa failed: 0x%x\n", status);
+    // Add new measurement.
+    if(derive_context(&derive_resp, DPE_DERIVE_CONTEXT_FLAG_RECURSIVE)) {
+        printf("Failed to add new measurement.\n");
+        return 1;
+    }
+
+    // Revoke existing exported CDI 
+    memcpy(&revoke_req.exported_cdi_handle, exported_resp.exported_cdi_handle, sizeof(uint8_t) * 32);
```

Comment:
nit: Can use `sizeof(revoke_req.exported_cdi_handle)` instead of computing manually